glib/src/boxed_any_object.rs

```diff
@@ -0,0 +1,165 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate as glib;
+use crate::subclass::prelude::*;
+use crate::{object_subclass, wrapper, Object};
+use std::any::Any;
+use std::cell::{Ref, RefCell, RefMut};
+
+#[derive(thiserror::Error, Debug)]
+pub enum BorrowError {
+    #[error("type of the inner value is not as requested")]
+    InvalidType,
+    #[error("value is already mutably borrowed")]
+    AlreadyBorrowed(#[from] std::cell::BorrowError),
+}
+#[derive(thiserror::Error, Debug)]
+pub enum BorrowMutError {
+    #[error("type of the inner value is not as requested")]
+    InvalidType,
+    #[error("value is already immutably borrowed")]
+    AlreadyMutBorrowed(#[from] std::cell::BorrowMutError),
+}
+
+mod imp {
+    use super::*;
+
+    #[derive(Debug)]
+    pub struct BoxedAnyObject {
+        pub value: RefCell<Box<dyn Any>>,
+    }
+    #[object_subclass]
+    impl ObjectSubclass for BoxedAnyObject {
+        const NAME: &'static str = "BoxedAnyObject";
+        type Type = super::BoxedAnyObject;
+    }
+    impl Default for BoxedAnyObject {
+        fn default() -> Self {
+            Self {
+                value: RefCell::new(Box::new(None::<usize>)),
+            }
+        }
+    }
+    impl ObjectImpl for BoxedAnyObject {}
+}
+
+wrapper! {
+    /// This is a subclass of `glib::object::Object` capable of storing any Rust type.
```

Comment:
```suggestion
    // rustdoc-stripper-ignore-next
    /// This is a subclass of `glib::object::Object` capable of storing any Rust type.
```

All the documentations comments, should be prefixed with rustdoc-stripper one. Otherwise they get removed with `python3 generate.py --strip-docs`. Normally we don't test that as part of the CI yet, but we will be doing so soon enough

Reply:
Not a problem in glib btw, it doesn't use `rustdoc-stripper`

Reply:
> Not a problem in glib btw, it doesn't use `rustdoc-stripper`

It definitely does :)

Reply:
Hrm, but we don't have the ignore-next before basically any of the docs :)

Reply:
If you rebase your changes on top of the master branch, the CI would properly complain now